FHIR-Converter/client/src/view/user-commands/convert.ts

```diff
@@ -15,52 +15,64 @@ import { showResultEditor } from '../common/editor/show-result-editor';
-	}
-	
-	// create the converter
-	const converter = ConverterEngineFactory.getInstance().createConverter();
+		// Get the data file
+		const dataFile = globals.settingManager.getWorkspaceState(stateConstants.DataKey);
+
+		// Check whether data file is dirty and not saved and ask if users want to save it
+		const doc = interaction.isDirtyFile(dataFile);
+		if (doc) {
+			await interaction.askSaveFiles([doc], localize('message.saveDataBeforeRefresh'), localize('message.save'), localize('message.ignore'));
+		}
+		
+		// create the converter
+		const converter = ConverterEngineFactory.getInstance().createConverter();
 
-	// Execute the conversion process
-	const resultFile = await converter.convert(dataFile);
+		// Execute the conversion process
+		const resultFile = await converter.convert(dataFile);
 
-	// Open the data in the editor
-	await vscode.window.showTextDocument(vscode.Uri.file(dataFile), {
-		viewColumn: vscode.ViewColumn.One
-	});
-	
-	// Open the template in the editor
-	const templateFile = globals.settingManager.getWorkspaceState(stateConstants.TemplateKey);
-	await vscode.window.showTextDocument(vscode.Uri.file(templateFile), {
-		viewColumn: vscode.ViewColumn.Two,
-	});
-	
-	// Obtain the enableDiffView option from the settings.
-	const enableDiff = globals.settingManager.getWorkspaceConfiguration(configurationConstants.enableDiffViewKey);
-	if (!enableDiff) {
-		await showResultEditor(vscode.Uri.file(resultFile));
-	} else {
-		// Get the history
-		const history = converter.getHistory(resultFile);
-		if (history.length === 1) {
-			// Show result in the editor
-			await showResultEditor(vscode.Uri.file(history[0]));
-		} else if (history.length > 1) {
-			// Show result with the differential view
-			await showDifferentialView(vscode.Uri.file(history[1]), vscode.Uri.file(history[0]));
+		// Open the data in the editor
+		await vscode.window.showTextDocument(vscode.Uri.file(dataFile), {
+			viewColumn: vscode.ViewColumn.One
+		});
+		
+		// Open the template in the editor
+		const templateFile = globals.settingManager.getWorkspaceState(stateConstants.TemplateKey);
+		await vscode.window.showTextDocument(vscode.Uri.file(templateFile), {
+			viewColumn: vscode.ViewColumn.Two,
+		});
+		
+		// Obtain the enableDiffView option from the settings.
+		const enableDiff = globals.settingManager.getWorkspaceConfiguration(configurationConstants.enableDiffViewKey);
+		if (!enableDiff) {
+			await showResultEditor(vscode.Uri.file(resultFile));
+		} else {
+			// Get the history
+			const history = converter.getHistory(resultFile);
+			if (history.length === 1) {
+				// Show result in the editor
+				await showResultEditor(vscode.Uri.file(history[0]));
+			} else if (history.length > 1) {
+				// Show result with the differential view
+				await showDifferentialView(vscode.Uri.file(history[1]), vscode.Uri.file(history[0]));
+			}
 		}
+		await vscode.commands.executeCommand('workbench.action.closeOtherEditors');
+	} catch (err) {
```

Comment:
does it possible don't catch this err? just use try/finally

Reply:
Thanks, updated it.